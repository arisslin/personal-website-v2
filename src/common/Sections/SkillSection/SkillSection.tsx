import * as React from 'react';
import styled from 'styled-components/macro';
import { breakpoints } from '../../styles/layout';
import SkillTag, { SkillTagProps } from '../../components/SkillTag/SkillTag';

export type SkillSectionProps = {
  tags: SkillTagProps[];
};

const SkillSection = ({ tags }: SkillSectionProps) => {
  if (tags.length === 0) {
    return <></>;
  }

  return (
    <Section className='layout-container'>
      <h2>Skills</h2>
      {tags.map((tag, index) => (
        <SkillTag key={`skill-tag-${index}`} text={tag.text} />
      ))}
    </Section>
  );
};

const Section = styled.section`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 10px;
  max-width: calc(${breakpoints.medium} - 4rem);
  margin: 0 auto;
`;

export default SkillSection;
