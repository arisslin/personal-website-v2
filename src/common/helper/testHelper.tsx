import { cleanup, fireEvent, render } from '@testing-library/react';

/**
 * Does a snapshot test for component it got.
 * @param {JSX.Element} component to creating snapshot for
 * @param {string} variant to create a postfix for the displayed test string
 * @return {void}
 */
export function doSnapshotTest(component: JSX.Element, variant?: string): void {
  const postfix = `${component.type.name as string} ${variant ? variant : ''}`;

  test(`Snapshot test for ${postfix}`, () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
    cleanup();
  });
}

/**
 * Fires an click event with 'fireEvent' from '@testing-library/react' on an
 * element.
 * @param {HTMLElement} element to fire click on.
 * @param {boolean} bubbles attribute of 'MouseEvent'.
 * @param {boolean} cancelable attribute of 'MouseEvent'.
 * @return {void}
 */
export function fireClickEvent(
  element: Element,
  bubbles: boolean,
  cancelable: boolean
) {
  fireEvent(element, new MouseEvent('click', { bubbles, cancelable }));
}
