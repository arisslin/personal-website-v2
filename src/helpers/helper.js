export const getText = (wantText) => {
  if (typeof wantText === 'boolean') {
    return wantText ? 'text' : null;
  }

  return null;
};
