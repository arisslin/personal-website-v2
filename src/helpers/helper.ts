export const getText = (wantText: boolean) => (wantText ? 'text' : null);
