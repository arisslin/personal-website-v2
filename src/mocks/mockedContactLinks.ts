import { ContactLinkProps } from '../common/components/ContactLink/ContactLink';
import { FaEnvelope, FaGithub, FaGlobe, FaXing } from 'react-icons/fa';

export const mockedContactLinks: ContactLinkProps[] = [
  {
    text: 'gueteklasse-a.de',
    href: 'https://www.gueteklasse-a.de',
    icon: FaGlobe,
  },
  {
    text: 'andreas.rissling@gmx.net',
    href: 'mailto:andreas.rissling@gmx.net',
    icon: FaEnvelope,
  },
  {
    text: 'GitHub',
    href: 'https://github.com/arisslin',
    icon: FaGithub,
  },
  {
    text: 'Xing',
    href: 'https://www.xing.com/profile/Andreas_Rissling3/',
    icon: FaXing,
  },
];
