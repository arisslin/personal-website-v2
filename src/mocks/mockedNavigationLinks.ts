import { Link } from '../types/dataTypes';

export const mockedNavigationLinks: Link[] = [
  { href: '', text: 'Home' },
  {
    href: '/aboutme',
    text: 'About me',
  },
];
