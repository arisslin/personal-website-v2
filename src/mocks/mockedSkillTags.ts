import { SkillTagProps } from '../common/components/SkillTag/SkillTag';

export const mockedSkillTags: SkillTagProps[] = [
  { text: 'HTML5' },
  { text: 'CSS3' },
  { text: 'JavaScript' },
  { text: 'TypeScript' },
  { text: 'React.jS' },
  { text: 'Gatsby' },
  { text: 'Node.js' },
  { text: 'Express.js' },
  { text: 'Bootstrap' },
  { text: 'Jest' },
  { text: 'Cypress' },
  { text: 'Storybook' },
  { text: 'Git' },
  { text: 'npm' },
  { text: 'agile Softwareentwicklung' },
  { text: 'Software Craftsmanship' },
  { text: 'TypeDoc' },
];
