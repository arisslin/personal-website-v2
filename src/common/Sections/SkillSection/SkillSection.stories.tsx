import * as React from 'react';
import { Meta, Story } from '@storybook/react';
import SkillSectionComponent, { SkillSectionProps } from './SkillSection';
import { mockedSkillTags } from '../../../mocks/mockedSkillTags';

export default {
  title: 'Sections/SkillSection',
  component: SkillSectionComponent,
  argTypes: {
    tags: {
      description: 'SkillTags which get rendered inside the section.',
    },
  },
  parameters: {
    docs: {
      description: {
        component: getDescription(),
      },
    },
  },
} as Meta;

export const SkillSection: Story<SkillSectionProps> = (args) => (
  <SkillSectionComponent tags={args.tags} />
);

SkillSection.args = {
  tags: mockedSkillTags,
};

function getDescription(): string {
  return (
    '<h2>About</h2>' +
    '<p>' +
    'The <b>SkillSection</b> got developed in Juli 2022. It renders a headline ' +
    'and some propped <b>SkillTags</b>.' +
    '</p>'
  );
}
