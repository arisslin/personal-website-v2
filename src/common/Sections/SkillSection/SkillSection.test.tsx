import * as React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import SkillSection from './SkillSection';
import { doSnapshotTest } from '../../helper/testHelper';
import { mockedSkillTags } from '../../../mocks/mockedSkillTags';

describe('SkillSection', () => {
  const component = <SkillSection tags={mockedSkillTags} />;

  afterEach(() => {
    cleanup();
  });

  it('renders with a headline h2', () => {
    render(component);

    const renderedComponent = screen.getByText('Skills');

    expect(renderedComponent).toBeInTheDocument();
    expect(renderedComponent.tagName).toBe('H2');
  });

  it('displays SkillTags', () => {
    render(component);

    mockedSkillTags.forEach((tag) => {
      expect(screen.getByText(tag.text)).toBeInTheDocument();
    });
  });

  it('does not render if tags are empty', () => {
    render(<SkillSection tags={[]} />);

    expect(screen.queryByText('Skills')).not.toBeInTheDocument();
  });

  doSnapshotTest(component);
});
