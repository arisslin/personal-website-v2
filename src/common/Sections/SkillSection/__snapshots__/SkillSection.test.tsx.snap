// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillSection Snapshot test for SkillSection  1`] = `
<div>
  <section
    class="SkillSection__Section-sc-cw10ze-0 ifurAI layout-container"
  >
    <h2>
      Skills
    </h2>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      HTML5
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      CSS3
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      JavaScript
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      TypeScript
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      React.jS
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Gatsby
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Node.js
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Express.js
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Bootstrap
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Jest
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Cypress
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Storybook
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Git
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      npm
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      agile Softwareentwicklung
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      Software Craftsmanship
    </span>
    <span
      class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
      data-testid="skill-tag"
    >
      TypeDoc
    </span>
  </section>
</div>
`;
